Homework/2/Program.cs

```diff
@@ -0,0 +1,470 @@
+        //
+        //Errors
+        //If there is no Users.txt file- the application should not throw "UsersNotFoundException.
+        //If a Users.text file contains duplicate users, throw a DuplicateUserCredentialsException.
+        //
+        //Restrictions
+        //Don't use File class static methods.
+
+
+
+        //Variables
+        private static bool loggedIn = false; // Variable to track if user is "logged in"
+        private static string[][] userArray = new string[0][]; // array of user arrays with login info which is string[username,password]; Would rather use List<>
+        private static string pathToFile = @"./users.txt"; // path to user file 
+
+        static void Main(string[] args)
+        {
+            LoadUsersFromFile();
+            MenuSelection();
+        }
+
+        //Main function which lists available options
+        static void MenuSelection() {
+            Console.Clear();
+            if (loggedIn)
+            {
+                Console.WriteLine("Logged in. \n\n");
+                Console.WriteLine("3 - List users\n");
+                Console.WriteLine("4 - Add user to start of list\n");
+                Console.WriteLine("5 - Add user to end of list\n");
+                Console.WriteLine("6 - Add user to n'th position in list\n");
+                Console.WriteLine("7 - Remove user from start of list\n");
+                Console.WriteLine("8 - Remove user from end of list\n");
+                Console.WriteLine("9 - Remove n'th user\n");
+                Console.WriteLine("10 - Sort users by username (asc/dec)\n");
+                Console.WriteLine("11 - Log out\n");
+            }
+            else {
+                Console.WriteLine("Select option:\n\n");
+                Console.WriteLine("1 - Log in\n");
+                Console.WriteLine("2 - Register\n");
+            }
+
+            Console.WriteLine("12 - Close console\n");
+
+            OptionSelection(option: Console.ReadLine());
+        }
+
+        // Loading users from file to array - HW3  / User details should follow this format: username;password - in different line for each user.
+        // Things to consider: line validation - multiple ;, empty usernames/passwords // ; char should also be validated on user creation
+        static void LoadUsersFromFile() {
+            try
+            {
+                using (StreamReader reader = new StreamReader(pathToFile))
+                {
+                    while (!reader.EndOfStream)
+                    {
+                        string[] userCredentials = reader.ReadLine().Split(';');
+                        if (CheckIfUserExists(userCredentials[0]))
+                        {
+                            throw new DuplicateUserCredentialsException(userCredentials[0]);
+                        }
+                        else
+                        {
+                            AddUserToArrayAtIndex(userCredentials, 0);
```

Comment:
This will put the users into `userCredentials` in reverse order.  It's not technically a bad thing but it seems like an unneeded complication.  Best to add each new user to the end of the array.